shell/models/cluster.x-k8s.io.machinedeployment.js

```diff
@@ -100,6 +101,21 @@ export default class CapiMachineDeployment extends SteveModel {
     return machinePools.find(pool => pool.machineConfigRef.name === machineConfigName);
   }
 
+  toggleScaleDownModal( event, showScalePoolPrompt, resources = this ) {
+    const alt = isAlternate(event);
+
+    if (!alt && !showScalePoolPrompt) {
+      this.$dispatch('promptModal', {
+        component:  'ScalePoolDownDialog',
+        resources,
+        modalWidth: '450px'
+      });
+    } else {
+      // User held alt key, so don't prompt
+      this.scalePool(-1);
```

Comment:
Is there not a way to pass this function to the modal as callback? It's inconvenient to define the logic twice.

Reply:
This will check if the user has pressed `Alt` while deleting pod in that case we will not show model

Reply:
Yes I know, but this logic is repeated in another file. Is it not possible to pass in the modal the function and define it once only?

Reply:
Since I've seen that it's a common patter, I set this to resolved.